### PR TITLE
Support Giving Suggestions in Comments

### LINKED ISSUE #33: Suggest Changes
It would be helpful to include a suggested fix in comments. We can support a list of fixes to apply in the configuration. For example:
```yaml
- diff_pattern: '\s+$'
  corrections:
    - find: '^(.*?)\s+$'
      replacement: '\1'
```

Need to support deleting the line (including the newline character).

This will be very useful for when we support running on local files as we can use the corrections to support automatically changing files.

### PR DESCRIPTION
**Pull request by juharris**
Resolves #33

### CODE REVIEW

File: src/config.py [hunk 1]
```diff
@@ -109,6 +130,12 @@ class Rule(TypedDict):
 	"""
 	vote: Optional[int | str]
 
+	# TODO Think about handling multiple suggestion blocks in one comment since right now all suggestions will create one block.
```
Comment (juharris) [inline src/config.py]: Probably not needed. People can just make a rule for each type of suggestion.

File: src/config.py [hunk 2]
```diff
@@ -49,6 +49,27 @@ class PolicyEvaluationChecks(TypedDict):
...
+	"""
+
+	pattern_regex: re.Pattern
+	""" Internal """
+
+	replacement: str
+	"""
+	The replacement for the pattern.
+	Python regex style substitution is supported.
+	For example: TODO Add examples.
```
Comment (outofthecave) [inline src/config.py]: Do you want to address this TODO in this PR or is this intentionally left for later?

File: src/suggestions/suggester.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		if suggestions is None:
+			return None
+
+		is_change_made = False
+		suggestion = text
+		while True:
+			found_matches = False
+			for suggestion_config in suggestions:
+				p = suggestion_config['pattern_regex']
+				while p.match(suggestion):
```
Comment (outofthecave) [inline src/suggestions/suggester.py]: `match` matches from the beginning of the string, but `sub` matches anywhere in the string.  You may want to use `search` here instead, which also matches anywhere in the string.

File: src/config.py (hunk 2 above)
Comment (juharris) [reply src/config.py]: Yeah I should come some example from the README.md

File: src/suggestions/suggester.py (hunk 3 above)
Comment (juharris) [reply src/suggestions/suggester.py]: Good point. I think you're right. I think the pattern should normally start with `^`, so it shouldn't matter, but I'll think more about this and make some tests to see what we need.

Comment (juharris) [reply src/suggestions/suggester.py]: I'll use subn so that we don't have to check for matches twice.

Comment (juharris) [reply src/suggestions/suggester.py]: I changed this to use `subn`. So it should be faster now and it lets us have more clear rules, like this one to remove trailing whitespace:
```yaml
    suggestions:
      - pattern: '\s+$',
        replacement: ''
```

**Pull request merged**